code/game/machinery/washing_machine.dm

```diff
@@ -14,306 +13,381 @@
+		color_sources += S
+	if(!length(color_sources))
+		return
+	color_source = pick(color_sources)
+	if(istype(color_source, /obj/item/toy/crayon))
+		var/obj/item/toy/crayon/C = color_source
+		wash_color = C.dye_color
+	if(istype(color_source, /obj/item/stamp))
+		var/obj/item/stamp/S = color_source
+		wash_color = S.dye_color
+
+/obj/machinery/washing_machine/proc/start_wash_cycle()
+	if(stat & (NOPOWER|BROKEN))
+		return
+	locate_color_source()
+	washing = TRUE
+	playsound(loc, 'sound/machines/click.ogg', 50, TRUE)
+	atom_say("Beginning Wash Cycle!")
+	sleep(1 SECONDS)
+	update_washing_state()
+	use_power(100)
+	playsound(loc, 'sound/effects/slosh.ogg', 25, TRUE)
+	addtimer(CALLBACK(src, PROC_REF(wash_cycle)), 15 SECONDS)
+
+/obj/machinery/washing_machine/proc/wash_cycle()
+	for(var/obj/item/I in inserted_items)
+		I.machine_wash(src)
+	for(var/mob/living/L in inserted_mobs)
+		L.machine_wash(src)
+	update_washing_state()
+	use_power(50 * current_tub_capacity)
+	playsound(loc, 'sound/effects/slosh.ogg', 25, TRUE)
+	addtimer(CALLBACK(src, PROC_REF(end_wash_cycle)), 15 SECONDS)
+
+/obj/machinery/washing_machine/proc/end_wash_cycle()
+	if(bloody_mess)
+		for(var/obj/item/I in inserted_items)
+			I.add_blood(I.blood_DNA)
+	if(color_source)
+		qdel(color_source)
+		color_source = null
+	playsound(loc, 'sound/effects/slosh.ogg', 25, TRUE)
+	washing = FALSE
+	atom_say("Wash Cycle Complete!")
+	sleep(1 SECONDS)
+	playsound(loc, 'sound/machines/ding.ogg', 50, TRUE)
+	update_washing_state()
+
+/obj/machinery/washing_machine/container_resist(mob/living/L)
+	. = ..()
+	if(washing)
+		return
+	if(!door_open)
+		toggle_door()
+		return
 	else
-		return ..()
+		L.forceMove(loc)
 
-/obj/machinery/washing_machine/attack_hand(mob/user)
-	switch(state)
-		if(OPEN_EMPTY)
-			state = CLOSED_EMPTY
-		if(CLOSED_EMPTY)
-			for(var/atom/movable/O in src)
-				O.forceMove(loc)
-			crayon = null
-			state = OPEN_EMPTY
-		if(OPEN_FULL)
-			state = CLOSED_FULL
-		if(CLOSED_FULL)
-			for(var/atom/movable/O in src)
-				O.forceMove(loc)
-			crayon = null
-			state = OPEN_EMPTY
-		if(RUNNING)
-			to_chat(user, "<span class='warning'>[src] is busy.</span>")
-		if(CLOSED_BLOODY)
-			if(gibs_ready)
-				gibs_ready = FALSE
-				if(locate(/mob) in src)
-					var/mob/M = locate() in src
-					M.gib()
-			for(var/atom/movable/O in src)
-				O.forceMove(loc)
-			crayon = null
-			state = OPEN_EMPTY
-
-	update_icon(UPDATE_ICON_STATE)
 
 /obj/machinery/washing_machine/deconstruct(disassembled = TRUE)
-	new /obj/item/stack/sheet/metal(drop_location(), 2)
+	eject_tub_contents()
+	new /obj/item/stack/sheet/metal(drop_location(), 5)
+	new /obj/item/circuitboard/washing_machine(drop_location(), 2)
+	qdel(src)
+
+/obj/machinery/washing_machine/AllowDrop()
+	return TRUE // so gibs and dropped items end up
+
+/obj/machinery/washing_machine/cleaning_act(mob/user, atom/cleaner, cleanspeed, text_verb, text_description, text_targetname)
+	. = ..()
+	if(.)
+		bloody_mess = FALSE
```

Comment:
Does this add a custom comment or do you need to add one?

Reply:
Yes, it produces a message. Also added update icons & did some check here aswell to make sure this works fully.